### PR TITLE
manual script instead of bindeps

### LINKED ISSUE #71: CPLEX install issue
~~~~~~~~~~~~~~
julia> Pkg.add("CPLEX")
INFO: Installing CPLEX v0.1.2
INFO: Building CPLEX

signal (11): Segmentation fault
unknown function (ip: 0x34f6a811a1)
__strdup at /lib64/libc.so.6 (unknown line)
jl_uv_dlopen at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
check_path! at /home/schalil/.julia/v0.4/BinDeps/src/dependencies.jl:558
jlcall_check_path!_21267 at  (unknown line)
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
_find_library at /home/schalil/.julia/v0.4/BinDeps/src/dependencies.jl:544
satisfy! at /home/schalil/.julia/v0.4/BinDeps/src/dependencies.jl:776
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
satisfy! at /home/schalil/.julia/v0.4/BinDeps/src/dependencies.jl:776
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
anonymous at /home/schalil/.julia/v0.4/BinDeps/src/dependencies.jl:831
unknown function (ip: 0x7ff0793009d3)
unknown function (ip: 0x7ff0793015ec)
jl_load at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
include at ./boot.jl:261
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
include_from_node1 at ./loading.jl:304
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
unknown function (ip: 0x7ff0792ec253)
unknown function (ip: 0x7ff0792eb5e9)
unknown function (ip: 0x7ff07930091c)
unknown function (ip: 0x7ff079300993)
jl_toplevel_eval_in at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
evalfile at loading.jl:320
evalfile at loading.jl:320
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
anonymous at none:14
cd at ./file.jl:22
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
anonymous at none:13
open at iostream.jl:114
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
unknown function (ip: 0x7ff0792ec253)
unknown function (ip: 0x7ff0792eb5e9)
unknown function (ip: 0x7ff07930091c)
unknown function (ip: 0x7ff079300993)
jl_toplevel_eval_in at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
process_options at ./client.jl:257
_start at ./client.jl:378
unknown function (ip: 0x7ff07157b8f9)
jl_apply_generic at /home/schalil/julia-cbe1bee3a8/bin/../lib/julia/libjulia.so (unknown line)
unknown function (ip: 0x401c47)
unknown function (ip: 0x40182f)
__libc_start_main at /lib64/libc.so.6 (unknown line)
unknown function (ip: 0x401875)
ERROR: Build process failed.
 in build! at pkg/entry.jl:701
 in build at pkg/entry.jl:720
 in resolve at ./pkg/entry.jl:479
 in edit at pkg/entry.jl:26
 in anonymous at task.jl:447
 in sync_end at ./task.jl:413
 [inlined code] from task.jl:422
 in add at pkg/entry.jl:46
 in add at pkg/entry.jl:73
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in add at pkg.jl:23
~~~~~~~~~~~~~~~~~~~~~~

I was trying to install CPLEX.jl. I have specified the path for CPLEX in bash_profile as below
~~~~~~~~~~~~~~~~~~~~
PATH=$PATH:$HOME/bin:$HOME/julia-cbe1bee3a8/bin:$HOME/CPLEX/
LD_LIBRARY_PATH="/home/schalil/CPLEX":$LD_LIBRARY_PATH

export PATH
export LD_LIBRARY_PATH
~~~~~~~~~~~~~~~~~~~~~
My cplex file location is as below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[schalil@user001 ~]$ cd CPLEX/
[schalil@user001 CPLEX]$ ls
cplex  libcplex1263.so
[schalil@user001 CPLEX]$ pwd
/home/schalil/CPLEX
[schalil@user001 CPLEX]$ 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
When I try to use CPLEX.jl, the error that i am getting is as below
~~~~~~~~~~~~~~~~~~~
julia> using CPLEX
INFO: Precompiling module CPLEX...
ERROR: LoadError: could not open file /home/schalil/.julia/v0.4/CPLEX/src/../deps/deps.jl
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:304
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:304
 [inlined code] from none:2
 in anonymous at no file:0
 in process_options at ./client.jl:257
 in _start at ./client.jl:378
while loading /home/schalil/.julia/v0.4/CPLEX/src/CPLEX.jl, in expression starting on line 8
ERROR: Failed to precompile CPLEX to /home/schalil/.julia/lib/v0.4/CPLEX.ji
 in error at ./error.jl:21
 in compilecache at loading.jl:384
 in require at ./loading.jl:250
~~~~~~~~~~~~~~~~~~~

### PR DESCRIPTION
**Pull request by mlubin**
I finally ran into the issue at #71, so time to dump BinDeps.
@joehuchette, I don't have a mac at the moment, could you help test this?

Still need to test on windows.

Closes #71 #15
CC @tkelman @schalil

### CODE REVIEW

**Comment by mlubin** [comment]
Also dropped cplex 12.4, don't think we should claim to support it.

File: deps/build.jl [hunk 1]
```diff
@@ -1,31 +1,57 @@
...
+import Compat: String
 
-cpxvers = ["124","125","1251","1260","1261","1262","1263"]
+depsfile = joinpath(dirname(@__FILE__),"deps.jl")
+if isfile(depsfile)
+    rm(depsfile)
+end
+
+function write_depsfile(path)
+    f = open(depsfile,"w")
```
Comment (tkelman) [inline deps/build.jl]: a do-block would be better

File: deps/build.jl [hunk 2]
```diff
@@ -1,31 +1,57 @@
...
+    close(f)
+end
 
-libnames = ASCIIString["cplex"]
-for v in cpxvers
-    push!(libnames, "cplex$v")
-    push!(libnames, "libcplex$v.so")
-    push!(libnames, "libcplex$v.dylib")
+if is_apple()
+	Libdl.dlopen("libstdc++",Libdl.RTLD_GLOBAL)
```
Comment (tkelman) [inline deps/build.jl]: I wonder whether this will be around forever. Will CPLEX on mac eventually move to using clang/libc++ ? Maybe they already have?

Comment (mlubin) [inline deps/build.jl]: Maybe @joehuchette can test on the latest release.

**Comment by joehuchette** [comment]
Works for me, both with and without the ``libstdc++`` block.

**Comment by mlubin** [comment]
> both with and without the ``libstdc++`` block.

That might only be for 12.6.3 though.

**Comment by joehuchette** [comment]
> That might only be for 12.6.3 though.

Well yeah, the code was presumably put there for some good reason.

**Comment by mlubin** [comment]
Just tested on Windows. I'll merge this and we'll test a bit more before tagging.

File: deps/build.jl [hunk 3]
```diff
@@ -1,31 +1,63 @@
...
-@BinDeps.setup
+using Compat
 
-@osx_only Libdl.dlopen("libstdc++",Libdl.RTLD_GLOBAL)
+import Compat: String
 
-cpxvers = ["124","125","1251","1260","1261","1262","1263"]
+depsfile = joinpath(dirname(@__FILE__),"deps.jl")
+if isfile(depsfile)
+    rm(depsfile)
```
Comment (tkelman) [inline deps/build.jl]: this is going to cause unnecessary recompilation, would be better to only write to the file if the content changes

**Pull request merged**